Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.py

```diff
@@ -0,0 +1,1322 @@
+        'product_description': result.get('product_description'),
+        'product_name': result.get('product_name'),
+        'product_version': result.get('product_version')
+    }
+
+    readable_output = tableToMarkdown('The file metadata', contents)
+    results = CommandResults(
+        outputs_prefix='CarbonBlackEEDR.File',
+        outputs_key_field='sha256',
+        outputs=context,
+        readable_output=readable_output,
+        raw_response=result
+    )
+    return results
+
+
+def get_file_command(client: Client, args: Dict) -> CommandResults:
+    sha256 = argToList(args.get('sha256'))
+    expiration_seconds = args.get('expiration_seconds')
+
+    result = client.get_file_request(sha256, expiration_seconds)
+    contents = []
+
+    found_files = result.get('found', [])
+    for file_ in found_files:
+        contents.append({
+            'sha256': file_.get('sha256'),
+            'url': file_.get('url')
+        })
+
+    readable_output = tableToMarkdown('The file to download', contents)
+    results = CommandResults(
+        outputs_prefix='CarbonBlackEEDR.File',
+        outputs_key_field='sha256',
+        outputs=result,
+        readable_output=readable_output,
+        raw_response=result
+    )
+    return results
+
+
+def get_file_path_command(client: Client, args: Dict) -> CommandResults:
+    sha256 = args.get('sha256')
+
+    result = client.get_file_path_request(sha256)
+    readable_output = tableToMarkdown('The file path for the sha256', result)
+    results = CommandResults(
+        outputs_prefix='CarbonBlackEEDR.File',
+        outputs_key_field='sha256',
+        outputs=result,
+        readable_output=readable_output,
+        raw_response=result
+    )
+    return results
+
+
+def download_file_to_xsoar_command(client: Client, args: Dict):
+    download_link = args.get('download_link')
+
+    result = client.get_download_file_content_values(download_link)
+    return result
+
+
+def fetch_incidents(client: Client, fetch_time: str, fetch_limit: str, last_run: Dict) -> Tuple[List, Dict]:
+    last_fetched_alert_create_time = last_run.get('last_fetched_alert_create_time')
+    last_fetched_alert_id = last_run.get('last_fetched_alert_id', '')
+    if not last_fetched_alert_create_time:
+        last_fetched_alert_create_time, _ = parse_date_range(fetch_time, date_format='%Y-%m-%dT%H:%M:%S.000Z')
+    latest_alert_create_date = last_fetched_alert_create_time
+    latest_alert_id = last_fetched_alert_id
+
+    incidents = []
+
+    response = client.search_alerts_request(
+        sort_field='first_event_time',
+        sort_order='ASC',
+        create_time=assign_params(
+            start=last_fetched_alert_create_time,
+            end=datetime_to_iso(datetime.now())
+        ),
+        limit=fetch_limit,
+    )
+    alerts = response.get('results', [])
+
+    for alert in alerts:
+        alert_id = alert.get('id')
+        if alert_id == last_fetched_alert_id:
```

Comment:
if i understand correctly, we always fetch an incidents we already fetched, and that will cause unexpected behaviour on some edge cases (such as limit set to 1, in which we will fetch 1 incident and wont fetch any new ones)
you can increase the `latest_alert_create_date` by 1 millisecond to overcome this